ssl-config/src/test/java/com/palantir/ssl/DropwizardServerTests.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.ssl;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import com.google.common.base.Optional;
+import com.palantir.remoting.http.FeignClients;
+import com.palantir.remoting.ssl.KeyStoreConfiguration;
+import com.palantir.remoting.ssl.SslConfiguration;
+import com.palantir.remoting.ssl.SslSocketFactories;
+import com.palantir.remoting.ssl.TrustStoreConfiguration;
+import feign.RetryableException;
+import io.dropwizard.Application;
+import io.dropwizard.Configuration;
+import io.dropwizard.setup.Environment;
+import io.dropwizard.testing.ConfigOverride;
+import io.dropwizard.testing.junit.DropwizardAppRule;
+import javax.net.ssl.SSLSocketFactory;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public final class DropwizardServerTests {
+    @ClassRule
+    public static final DropwizardAppRule<Configuration> APP = new DropwizardAppRule<>(
+            TestEchoServer.class,
+            "src/test/resources/test-server.yml",
+            ConfigOverride.config("server.applicationConnectors[0].keyStorePath",
+                    TestConstants.SERVER_KEY_STORE_JKS_PATH.toString()),
+            ConfigOverride.config("server.applicationConnectors[0].keyStorePassword",
+                    TestConstants.SERVER_KEY_STORE_JKS_PASSWORD),
+            ConfigOverride.config("server.applicationConnectors[0].trustStorePath",
+                    TestConstants.CA_TRUST_STORE_PATH.toString()),
+            ConfigOverride.config("server.applicationConnectors[0].crlPath",
+                    TestConstants.CA_CRL_PATH.toString()),
+            ConfigOverride.config("server.applicationConnectors[0].needClientAuth",
+                    "true"));
+
+    @Test
+    public void testConnectionFailsWithoutClientCerts() {
+        SslConfiguration sslConfig = SslConfiguration.of(TrustStoreConfiguration.of(TestConstants.CA_TRUST_STORE_PATH));
+        TestEchoService service = createTestService(sslConfig);
+
+        try {
+            service.echo("foo");
+            fail();
+        } catch (RetryableException ex) {
+            // expected
```

Comment:
Originally had this verify specific exception and message, but it would pass locally and fail in the build system (presumably because the exception on the build different was a different one than what I was seeing locally). Unfortunately, I couldn't figure out how to get the full build log/make the output more granular, so I couldn't customize the check. If someone can show me how to get more detailed failure logs, I can update this test to make it more specific.